File: example/lib/android/runtimestub.dart
import 'package:dart_native/dart_native.dart';

class RuntimeStub extends JObject {
  RuntimeStub() : super("com/dartnative/dart_native_example/RuntimeStub");


  double getFloat() {
    return invoke('getFloat', [8.65], true);
  }

  double getDouble() {
    return invoke('getDouble', [10.22]);
  }

  String getChar() {
    return invoke('getChar', ['a']);
  }

  int getInt() {
    return invoke('getInt', [10]);
  }

  bool getBool() {
    return invoke('getBool', [true]);
  }

  JObject getObject(JObject object) {
    return invoke('getObject', [object]);
  }

//  int getByte() {
//    return invoke('getByte');
//  }
//
//  int getShort() {
//    return invoke('getShort');
//  }
//
//  int getLong() {
//    return invoke('getLong');
//  }
//
//  int getChar() {
//    return invoke('getChar');
//  }
}

File: example/android/app/src/main/java/com/dartnative/dart_native_example/RuntimeStub.java
package com.dartnative.dart_native_example;

import androidx.annotation.NonNull;
import io.flutter.Log;

public class RuntimeStub {
  public static final String TAG = "dart_java";

  public static int getInt(int i){
    Log.d(TAG, "getInt : " + i);
    return 100;
  }

  public static double getDouble(double b) {
    Log.d(TAG, "getDouble : " + b);
    return 100.23;
  }

  public static byte getByte() {
    return 1;
  }

  public static float getFloat(float f) {
    Log.d(TAG, "getFloat : " + f);
    return 9.8f;
  }

  public static char getChar(char c) {
    Log.d(TAG, "getChar : " + c);
    return 'b';
  }

  public static short getShort() {
    return 1;
  }

  public static long getLong() {
    return 1;
  }

  public static boolean getBool(boolean b) {
    Log.d(TAG, "getBool : " + b);
    return false;
  }
}
